### PR TITLE
feat: harmonise sc-element types

### PR DESCRIPTION
**Pull request by NikitaZotov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -335,15 +335,15 @@ link_node: ScAddr
...
+    sc_type.VAR_NODE_LINK >> '_link',
+    sc_type.VAR_PERM_POS_ARC,
     relation_node,
 )
 template.triple(
     '_main_node',
-    sc_types.EDGE_ACCESS_VAR_POS_TEMP,
-    (sc_types.NODE_VAR, '_var_node')
+    sc_type.VAR_TEMP_POS_ARC,
+    (sc_type.VAR_NODE, '_var_node')
```
Comment (kilativ-dotcom) [inline README.md]: shouldn't is be with >>?

File: README.md [hunk 2]
```diff
@@ -451,11 +451,11 @@ Delete *addrs* from the KB memory and returns boolean status.
...
-from sc_client.constants import sc_types
+from sc_client.constants import sc_type
 from sc_client.models import ScConstruction, ScLinkContent, ScLinkContentType
 
 construction = ScConstruction()  # Create link for example
-construction.generate_link(sc_types.LINK_CONST, ScLinkContent("One", ScLinkContentType.STRING))
+construction.generate_link(sc_type.CONST_NODE_LINK, ScLinkContent("One", ScLinkContentType.STRING))
 link = generate_elements(construction)[0]
 
 link_content = ScLinkContent("Two", ScLinkContentType.STRING, link)
```
Comment (kilativ-dotcom) [inline README.md]: 'Erase elements' section does not contain any erase_elements examples

File: docs/changelog.md [hunk 3]
```diff
@@ -40,7 +40,62 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
   New sc-event types: `AFTER_GENERATE_CONNECTOR`, `AFTER_GENERATE_EDGE`, `BEFORE_ERASE_CONNECTOR`, `BEFORE_ERASE_EDGE` were added.
 
+ - All sc-types were also redesigned to a common style. They were deprecated, new ones were added. See changes in the table below.
```
Comment (kilativ-dotcom) [inline docs/changelog.md]:
```suggestion
 - All sc-types were redesigned to a common style. They were deprecated, new ones were added. See changes in the table below.
```

File: docs/changelog.md [hunk 4]
```diff
@@ -40,7 +40,62 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  | sc_types.LINK_VAR                   | sc_type.VAR_NODE_LINK         |
+  | sc_types.LINK_VAR_CLASS             | sc_type.VAR_NODE_LINK_CLASS   |
+  | sc_types.NODE_VAR_STRUCT            | sc_type.VAR_NODE_STRUCTURE    |
+  | sc_types.NODE_VAR_TUPLE             | sc_type.VAR_NODE_TUPLE        |
+  | sc_types.NODE_VAR_ROLE              | sc_type.VAR_NODE_ROLE         |
+  | sc_types.NODE_VAR_NO_ROLE           | sc_type.VAR_NODE_NO_ROLE      |
+  | sc_types.NODE_VAR_CLASS             | sc_type.VAR_NODE_CLASS        |
+  | sc_types.NODE_VAR_MATERIAL          | sc_type.VAR_NODE_MATERIAL     | 
+  
+  From now on, all sc-links are sc-nodes. Types of actual and inactual temporal membership sc-arc were added.
```
Comment (kilativ-dotcom) [inline docs/changelog.md]:
```suggestion
  - All sc-links are sc-nodes.
  - Types of actual and inactual temporal membership sc-arc were added.
```

File: docs/changelog.md [hunk 5]
```diff
@@ -40,7 +40,62 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  Module `sc_types` is deprecated, use `sc_type` instead.
+
+- Type `ScType.NodeAbstract` was removed.
+
 ### Added
+ - ScType methods: `is_connector`, `is_structure`, `is_no_role`, `is_actual`, `is_inactual`, `is_superclass`
+ - All possible combinations of subtypes into sc-types
+ - Type `sc_type.NODE_SUPERCLASS`
+ - Types: `sc_type.CONNECTOR`, `sc_type.ARC`
+ - Types of actual and inactual temporal sc-arcs into sc-types
```
Comment (kilativ-dotcom) [inline docs/changelog.md]:
```suggestion
 - All possible combinations of subtypes into sc-type
 - Type `sc_type.NODE_SUPERCLASS`
 - Types: `sc_type.CONNECTOR`, `sc_type.ARC`
 - Types of actual and inactual temporal sc-arcs into sc-type
```

File: docs/changelog.md [hunk 6]
```diff
@@ -40,7 +40,62 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  | sc_types.LINK_VAR_CLASS             | sc_type.VAR_NODE_LINK_CLASS   |
+  | sc_types.NODE_VAR_STRUCT            | sc_type.VAR_NODE_STRUCTURE    |
+  | sc_types.NODE_VAR_TUPLE             | sc_type.VAR_NODE_TUPLE        |
+  | sc_types.NODE_VAR_ROLE              | sc_type.VAR_NODE_ROLE         |
+  | sc_types.NODE_VAR_NO_ROLE           | sc_type.VAR_NODE_NO_ROLE      |
+  | sc_types.NODE_VAR_CLASS             | sc_type.VAR_NODE_CLASS        |
+  | sc_types.NODE_VAR_MATERIAL          | sc_type.VAR_NODE_MATERIAL     | 
+  
+  From now on, all sc-links are sc-nodes. Types of actual and inactual temporal membership sc-arc were added.
+  Module `sc_types` is deprecated, use `sc_type` instead.
```
Comment (kilativ-dotcom) [inline docs/changelog.md]:
```suggestion
  - Module `sc_types` is deprecated, use `sc_type` instead.
```

File: docs/changelog.md [hunk 7]
```diff
@@ -49,11 +104,20 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
  - `ScEvent` class was renamed to `ScEventSubscription`
  - `ScEventParams` class was renamed to `ScEventSubscriptionParams`
 
+### Fixed
+ - Checking of all syntactic and semantic subtypes for types in ScType `merge` method.
+ - Now sc-link is sc-node
```
Comment (kilativ-dotcom) [inline docs/changelog.md:109]:
```suggestion
 - sc-link is sc-node
```

File: src/sc_client/constants/sc_type/__init__.py [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+CONST_ACTUAL_TEMP_ARC = ScType(b.SC_TYPE_CONST | b.SC_TYPE_ACTUAL_ARC | b.SC_TYPE_TEMP_ARC)
+VAR_ACTUAL_TEMP_ARC = ScType(b.SC_TYPE_VAR | b.SC_TYPE_ACTUAL_ARC | b.SC_TYPE_TEMP_ARC)
+CONST_INACTUAL_TEMP_ARC = ScType(b.SC_TYPE_CONST | b.SC_TYPE_INACTUAL_ARC | b.SC_TYPE_TEMP_ARC)
+VAR_INACTUAL_TEMP_ARC = ScType(b.SC_TYPE_VAR | b.SC_TYPE_INACTUAL_ARC | b.SC_TYPE_TEMP_ARC)
+
+# positivity
+POS_ARC = ScType(b.SC_TYPE_POS_ARC)
+NEG_ARC = ScType(b.SC_TYPE_NEG_ARC)
+
+# fuzzy sc-arcs
```
Comment (kilativ-dotcom) [inline src/sc_client/constants/sc_type/__init__.py]:
```suggestion
# fuzziness
```

File: src/sc_client/constants/sc_type/bitmasks.py [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+SC_TYPE_INACTUAL_ARC = SC_TYPE_MEMBERSHIP_ARC | 0x2000
+
+# sc-arc permanence
+SC_TYPE_TEMP_ARC = SC_TYPE_MEMBERSHIP_ARC | 0x400
+SC_TYPE_PERM_ARC = SC_TYPE_MEMBERSHIP_ARC | SC_TYPE_ACTUAL_ARC | 0x800
+
+# sc-arc positivity
+SC_TYPE_POS_ARC = SC_TYPE_MEMBERSHIP_ARC | 0x80
+SC_TYPE_NEG_ARC = SC_TYPE_MEMBERSHIP_ARC | 0x100
+SC_TYPE_FUZ_ARC = SC_TYPE_MEMBERSHIP_ARC | 0x200
```
Comment (kilativ-dotcom) [inline src/sc_client/constants/sc_type/bitmasks.py:27]: fuz is not positivity arc

File: src/sc_client/constants/sc_type/bitmasks.py [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+# TYPE MASK
+SC_TYPE_ELEMENT_MASK = SC_TYPE_NODE | SC_TYPE_CONNECTOR
+SC_TYPE_CONNECTOR_MASK = SC_TYPE_COMMON_EDGE | SC_TYPE_COMMON_ARC | SC_TYPE_MEMBERSHIP_ARC
+SC_TYPE_ARC_MASK = SC_TYPE_COMMON_ARC | SC_TYPE_MEMBERSHIP_ARC
+
+SC_TYPE_CONSTANCY_MASK = SC_TYPE_CONST | SC_TYPE_VAR
+SC_TYPE_ACTUALITY_MASK = SC_TYPE_ACTUAL_ARC | SC_TYPE_INACTUAL_ARC
+SC_TYPE_PERMANENCY_MASK = SC_TYPE_PERM_ARC | SC_TYPE_TEMP_ARC
+SC_TYPE_POSITIVITY_MASK = SC_TYPE_POS_ARC | SC_TYPE_NEG_ARC | SC_TYPE_FUZ_ARC
```
Comment (kilativ-dotcom) [inline src/sc_client/constants/sc_type/bitmasks.py]: fuz is not positivity

File: src/sc_client/constants/sc_type/sc_type.py [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+import warnings
+
+from sc_client.constants.exceptions import InvalidTypeError
+from sc_client.constants.sc_type import bitmasks
+
+
+class ScType:
+    def __init__(self, value: int = 0):
+        if not isinstance(value, (ScType, int)):
+            raise InvalidTypeError("You should to use int or ScType type for ScType initialization")
```
Comment (kilativ-dotcom) [inline src/sc_client/constants/sc_type/sc_type.py]:
```suggestion
            raise InvalidTypeError("You should use int or ScType type for ScType initialization")
```

File: src/sc_client/constants/sc_type/sc_type.py [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+        return (self.value & bitmasks.SC_TYPE_PERM_ARC) == bitmasks.SC_TYPE_PERM_ARC
+
+    def is_temp(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_TEMP_ARC) == bitmasks.SC_TYPE_TEMP_ARC
+
+    def is_actual(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_ACTUAL_ARC) == bitmasks.SC_TYPE_ACTUAL_ARC
+
+    def is_inactual(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_INACTUAL_ARC) == bitmasks.SC_TYPE_INACTUAL_ARC
```
Comment (kilativ-dotcom) [inline src/sc_client/constants/sc_type/sc_type.py]: why are some methods comparing result with 0 and some are comparing result with type? also a lot of duplicated code, can be in some method that takes type and returns bool

File: src/sc_client/constants/sc_type/sc_type.py [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+
+    def is_equal(self, other: ScType) -> bool:
+        return self.__eq__(other)
+    
+    def _is_not_compatible_by_mask(self, new_type: ScType, mask):
+        subtype = self.value & mask
+        new_subtype = new_type.value & mask
+        return subtype != bitmasks.SC_TYPE_UNKNOWN and subtype != new_subtype
+    
+    def _is_expendable_to(self, new_type: ScType): # it is equal to `sc_storage_is_type_expendable_to` in the sc-machine
```
Comment (kilativ-dotcom) [inline src/sc_client/constants/sc_type/sc_type.py]: use extend instead of expend everywhere

File: src/sc_client/constants/sc_type/sc_type.py [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+        self_value = self.value
+        new_value = new_type.value
```
Comment (kilativ-dotcom) [inline src/sc_client/constants/sc_type/sc_type.py:137]: these two aren't needed here
later they are assigned to a variable of type ScType, so their name is not correct as well

File: src/sc_client/constants/sc_type/sc_type.py [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+    def is_valid(self) -> bool:
+        return self.__bool__()
+
+    def is_equal(self, other: ScType) -> bool:
+        return self.__eq__(other)
+    
+    def _is_not_compatible_by_mask(self, new_type: ScType, mask):
+        subtype = self.value & mask
+        new_subtype = new_type.value & mask
+        return subtype != bitmasks.SC_TYPE_UNKNOWN and subtype != new_subtype
```
Comment (kilativ-dotcom) [inline src/sc_client/constants/sc_type/sc_type.py]: this method can be static, it is mostly used right after constructing two new ScType variables but it is not necessary, ints can be passed here

File: tests/models_test.py [hunk 16]
```diff
@@ -357,11 +621,18 @@ def test_change_const(self):
         assert t.CONST.change_const(False).is_var()
 
     def test_merge(self):
-        assert t.NODE_CONST.merge(t.NODE_CLASS) == t.NODE_CONST_CLASS
+        assert t.UNKNOWN.merge(t.NODE) == t.NODE
+        assert t.NODE.merge(t.CONST_NODE) == t.CONST_NODE
+        assert t.CONST_NODE.merge(t.CONST_NODE_CLASS) == t.CONST_NODE_CLASS
+        assert t.NODE.merge(t.NODE_LINK_CLASS) == t.NODE_LINK_CLASS
+        assert t.CONNECTOR.merge(t.CONST_MEMBERSHIP_ARC) == t.CONST_MEMBERSHIP_ARC
         with pytest.raises(InvalidTypeError, match=CommonErrorMessages.INVALID_TYPE.value):
```
Comment (kilativ-dotcom) [inline tests/models_test.py:619]: there is a chance that only first statement in this block is executed, can you split it into multiple `with pytest.raises` blocks?

**Review by kilativ-dotcom — CHANGES_REQUESTED** [review]

File: src/sc_client/constants/sc_type/__init__.py (hunk 8 above)
Comment (kilativ-dotcom) [reply src/sc_client/constants/sc_type/__init__.py]: and place it after // positive sc-arcs section to follow common style when new subtype is introduced and then it is combined with all possible types

File: README.md (hunk 1 above)
Comment (NikitaZotov) [reply README.md]: You can use () or >>

File: README.md (hunk 2 above)
Comment (NikitaZotov) [reply README.md]: It has nothing to do with this PR

File: docs/changelog.md (hunk 4 above)
Comment (NikitaZotov) [reply docs/changelog.md]: okay

File: docs/changelog.md (hunk 6 above)
Comment (NikitaZotov) [reply docs/changelog.md]: Okay

File: docs/changelog.md (hunk 7 above)
Comment (NikitaZotov) [reply docs/changelog.md:109]: No

File: src/sc_client/constants/sc_type/bitmasks.py (hunk 9 above)
Comment (NikitaZotov) [reply src/sc_client/constants/sc_type/bitmasks.py:27]: Okay

File: src/sc_client/constants/sc_type/__init__.py (hunk 8 above)
Comment (NikitaZotov) [reply src/sc_client/constants/sc_type/__init__.py]: I'm not quite sure where you want to move it

File: src/sc_client/constants/sc_type/sc_type.py (hunk 12 above)
Comment (NikitaZotov) [reply src/sc_client/constants/sc_type/sc_type.py]: okay

File: src/sc_client/constants/sc_type/sc_type.py (hunk 13 above)
Comment (NikitaZotov) [reply src/sc_client/constants/sc_type/sc_type.py]: okay

File: src/sc_client/constants/sc_type/sc_type.py (hunk 14 above)
Comment (NikitaZotov) [reply src/sc_client/constants/sc_type/sc_type.py:137]: There are needed, see below

File: src/sc_client/constants/sc_type/sc_type.py (hunk 15 above)
Comment (NikitaZotov) [reply src/sc_client/constants/sc_type/sc_type.py]: What are the advantages of the static method over the current way? It doesn't provide anything here

File: README.md (hunk 1 above)
Comment (kilativ-dotcom) [reply README.md]: but shouldn't documentation be written in a way so user's won't question when they have to use () and when they have to use >>? I think it should use one style across all examples

File: docs/changelog.md (hunk 5 above)
Comment (kilativ-dotcom) [reply docs/changelog.md]: there were 2 suggestions in this comment

File: docs/changelog.md (hunk 7 above)
Comment (kilativ-dotcom) [reply docs/changelog.md:109]: all entries in changelog mean that it is already in the project, why don't you add 'now' to all of them then?

File: src/sc_client/constants/sc_type/__init__.py (hunk 8 above)
Comment (kilativ-dotcom) [reply src/sc_client/constants/sc_type/__init__.py]: before line that starts with '# fuzzy sc-arcs'

File: src/sc_client/constants/sc_type/bitmasks.py [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+SC_TYPE_ELEMENT_MASK = SC_TYPE_NODE | SC_TYPE_CONNECTOR
+SC_TYPE_CONNECTOR_MASK = SC_TYPE_COMMON_EDGE | SC_TYPE_COMMON_ARC | SC_TYPE_MEMBERSHIP_ARC
+SC_TYPE_ARC_MASK = SC_TYPE_COMMON_ARC | SC_TYPE_MEMBERSHIP_ARC
+
+SC_TYPE_CONSTANCY_MASK = SC_TYPE_CONST | SC_TYPE_VAR
+SC_TYPE_ACTUALITY_MASK = SC_TYPE_ACTUAL_ARC | SC_TYPE_INACTUAL_ARC
+SC_TYPE_PERMANENCY_MASK = SC_TYPE_PERM_ARC | SC_TYPE_TEMP_ARC
+SC_TYPE_POSITIVITY_MASK = SC_TYPE_POS_ARC | SC_TYPE_NEG_ARC
+
+SC_TYPE_MEMBERSHIP_ARC_MASK = SC_TYPE_ACTUALITY_MASK | SC_TYPE_PERMANENCY_MASK | SC_TYPE_POSITIVITY_MASK
```
Comment (kilativ-dotcom) [inline src/sc_client/constants/sc_type/bitmasks.py]: now fuzzy arcs are not membership arcs?

File: src/sc_client/constants/sc_type/sc_type.py [hunk 18]
```diff
@@ -0,0 +1,203 @@
...
+                return False
+
+            if self_type._is_not_compatible_by_mask(new_type, bitmasks.SC_TYPE_POSITIVITY_MASK):
+                return False
+
+        return True
+
+    def merge(self, other: ScType) -> ScType:
+        if not self._is_extendable_to(other):
+            raise InvalidTypeError(f"Type `{self}` can not be expended to `{other}`.")
```
Comment (kilativ-dotcom) [inline src/sc_client/constants/sc_type/sc_type.py]:
```suggestion
            raise InvalidTypeError(f"Type `{self}` can not be extended to `{other}`.")
```

File: src/sc_client/constants/sc_type/sc_type.py (hunk 15 above)
Comment (kilativ-dotcom) [reply src/sc_client/constants/sc_type/sc_type.py]: it is always called right after constructing of new objects, so it'll use less statements to do same task
image

File: README.md (hunk 1 above)
Comment (NikitaZotov) [reply README.md]: Okay

File: README.md (hunk 2 above)
Comment (NikitaZotov) [reply README.md]: Okay

File: docs/changelog.md (hunk 5 above)
Comment (NikitaZotov) [reply docs/changelog.md]: I removed the second `in sc-types`

File: docs/changelog.md (hunk 7 above)
Comment (NikitaZotov) [reply docs/changelog.md:109]: `sc-link is sc-node` is not understandable mention for users

File: src/sc_client/constants/sc_type/sc_type.py (hunk 15 above)
Comment (NikitaZotov) [reply src/sc_client/constants/sc_type/sc_type.py]: `it'll use less statements to do same task` What do you mean?

File: tests/models_test.py (hunk 16 above)
Comment (NikitaZotov) [reply tests/models_test.py:619]: it was fixed. Resee it, please

File: src/sc_client/constants/sc_type/__init__.py (hunk 8 above)
Comment (NikitaZotov) [reply src/sc_client/constants/sc_type/__init__.py]: I still don't understand what you want me to do.

File: src/sc_client/constants/sc_type/sc_type.py (hunk 15 above)
Comment (kilativ-dotcom) [reply src/sc_client/constants/sc_type/sc_type.py]: Constructors and destructors won't be called so at least 4 statements per check will not be executed
Ints can be compared in static method without the creation of ScTypes

Comment (NikitaZotov) [reply src/sc_client/constants/sc_type/sc_type.py]: What does that do?

**Review by kilativ-dotcom — CHANGES_REQUESTED** [review]
fix tests

File: docs/changelog.md (hunk 5 above)
Comment (NikitaZotov) [reply docs/changelog.md]: Fixed

File: src/sc_client/constants/sc_type/__init__.py (hunk 8 above)
Comment (NikitaZotov) [reply src/sc_client/constants/sc_type/__init__.py]: Fixed

File: src/sc_client/constants/sc_type/sc_type.py (hunk 15 above)
Comment (NikitaZotov) [reply src/sc_client/constants/sc_type/sc_type.py]: Fixed, ok

**Review by kilativ-dotcom — APPROVED** [review]

**Review by kilativ-dotcom — CHANGES_REQUESTED** [review]
use non_role_relation

File: docs/changelog.md [hunk 19]
```diff
@@ -40,7 +40,62 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  | sc_types.NODE_CONST_ROLE            | sc_type.CONST_NODE_ROLE       |
+  | sc_types.NODE_CONST_NON_ROLE         | sc_type.CONST_NODE_NON_ROLE    |
+  | sc_types.NODE_CONST_CLASS           | sc_type.CONST_NODE_CLASS      |
+  | sc_types.NODE_CONST_MATERIAL        | sc_type.CONST_NODE_MATERIAL   |
+  | sc_types.LINK_VAR                   | sc_type.VAR_NODE_LINK         |
+  | sc_types.LINK_VAR_CLASS             | sc_type.VAR_NODE_LINK_CLASS   |
+  | sc_types.NODE_VAR_STRUCT            | sc_type.VAR_NODE_STRUCTURE    |
+  | sc_types.NODE_VAR_TUPLE             | sc_type.VAR_NODE_TUPLE        |
+  | sc_types.NODE_VAR_ROLE              | sc_type.VAR_NODE_ROLE         |
+  | sc_types.NODE_VAR_NON_ROLE           | sc_type.VAR_NODE_NON_ROLE      |
```
Comment (kilativ-dotcom) [inline docs/changelog.md]: deprecated should be NO_ROLE, but you made it NON_ROLE

File: docs/changelog.md [hunk 20]
```diff
@@ -49,11 +104,20 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
...EDGE_U_COMMON`, `sc_types.EDGE_D_COMMON`, `sc_types.EDGE_U_COMMON_CONST`, `sc_types.EDGE_D_COMMON_CONST`, `sc_types.EDGE_ACCESS`, `sc_types.EDGE_ACCESS_CONST_POS_PERM`, `sc_types.EDGE_ACCESS_CONST_NEG_PERM`, `sc_types.EDGE_ACCESS_CONST_FUZ_PERM`, `sc_types.EDGE_ACCESS_CONST_POS_TEMP`, `sc_types.EDGE_ACCESS_CONST_NEG_TEMP`, `sc_types.EDGE_ACCESS_CONST_FUZ_TEMP`, `sc_types.EDGE_U_COMMON_VAR`, `sc_types.EDGE_D_COMMON_VAR`, `sc_types.EDGE_ACCESS_VAR_POS_PERM`, `sc_types.EDGE_ACCESS_VAR_NEG_PERM`, `sc_types.EDGE_ACCESS_VAR_FUZ_PERM`, `sc_types.EDGE_ACCESS_VAR_POS_TEMP`, `sc_types.EDGE_ACCESS_VAR_NEG_TEMP`, `sc_types.EDGE_ACCESS_VAR_FUZ_TEMP`, `sc_types.NODE_CONST`, `sc_types.NODE_VAR`, `sc_types.LINK`, `sc_types.LINK_CLASS`, `sc_types.NODE_STRUCT`, `sc_types.LINK_CONST`, `sc_types.LINK_CONST_CLASS`, `sc_types.NODE_CONST_TUPLE`, `sc_types.NODE_CONST_STRUCT`, `sc_types.NODE_CONST_ROLE`, `sc_types.NODE_CONST_NON_ROLE`, `sc_types.NODE_CONST_CLASS`, `sc_types.NODE_CONST_MATERIAL`, `sc_types.LINK_VAR`, `sc_types.LINK_VAR_CLASS`, `sc_types.NODE_VAR_STRUCT`, `sc_types.NODE_VAR_TUPLE`, `sc_types.NODE_VAR_ROLE`, `sc_types.NODE_VAR_NON_ROLE`, `sc_types.NODE_VAR_CLASS`, `sc_types.NODE_VAR_MATERIAL`
```
Comment (kilativ-dotcom) [inline docs/changelog.md]: non role is in this list

**Review by kilativ-dotcom — CHANGES_REQUESTED** [review]

File: docs/changelog.md [hunk 21]
```diff
@@ -40,7 +40,62 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  | sc_types.NODE_CONST_ROLE            | sc_type.CONST_NODE_ROLE       |
+  | sc_types.NODE_CONST_NO_ROLE         | sc_type.CONST_NODE_NON_ROLE   |
+  | sc_types.NODE_CONST_CLASS           | sc_type.CONST_NODE_CLASS      |
+  | sc_types.NODE_CONST_MATERIAL        | sc_type.CONST_NODE_MATERIAL   |
+  | sc_types.LINK_VAR                   | sc_type.VAR_NODE_LINK         |
+  | sc_types.LINK_VAR_CLASS             | sc_type.VAR_NODE_LINK_CLASS   |
+  | sc_types.NODE_VAR_STRUCT            | sc_type.VAR_NODE_STRUCTURE    |
+  | sc_types.NODE_VAR_TUPLE             | sc_type.VAR_NODE_TUPLE        |
+  | sc_types.NODE_VAR_ROLE              | sc_type.VAR_NODE_ROLE         |
+  | sc_types.NODE_VAR_NO_ROLE           | sc_type.VAR_NODE_NON_ROLE      |
```
Comment (kilativ-dotcom) [inline docs/changelog.md]:
```suggestion
  | sc_types.NODE_VAR_NO_ROLE           | sc_type.VAR_NODE_NON_ROLE     |
```

**Review by kilativ-dotcom — APPROVED** [review]

**Pull request merged**